scanner/src/main/kotlin/Scanner.kt

```diff
@@ -81,8 +81,11 @@ abstract class Scanner {
         }
 
         val sourceDirectory = try {
-            val downloadDirectory = (downloadDirectory ?: File(outputDirectory, "downloads")).apply { safeMkdirs() }
-            Main.download(pkg, downloadDirectory)
+            if (downloadDirectory != null) {
+                Main.download(pkg, downloadDirectory)
+            } else {
+                Main.download(pkg, File(outputDirectory, "downloads"))
+            }
```

Comment:
This could also go to one line:
`Main.download(pkg, downloadDirectory ?: File(outputDirectory, "downloads"))`

Reply:
Done.